### PR TITLE
add options to nanopore uploads

### PR DESCRIPTION
**Pull request by morsecodist**
- Add Midnight wetlab protocol
- Add ARTIC v4 wetlab protocol
- Add medaka model options
- Allow wetlab protocols for nanopore

### CODE REVIEW

File: cmd/consensusGenome/consensusGenome.go [hunk 1]
```diff
@@ -25,16 +27,53 @@ var Technologies = map[string]string{
 var technologyOptionsString string
 
 var WetlabProtocols = map[string]string{
```
Comment (tfrcarvalho) [inline cmd/consensusGenome/consensusGenome.go:29]: More as follow up... this cli hard code does not seem sustainable. Should we to create API endpoints for this info?

File: cmd/consensusGenome/consensusGenome.go [hunk 2]
```diff
@@ -82,8 +146,25 @@ func validateCommonArgs() error {
...
+		wetlabProtocol = "ARTIC v3"
+	}
+	if _, has := nanoporeWetLabProtocols[wetlabProtocol]; wetlabProtocol != "" && technology == "Nanopore" && !has {
+		return fmt.Errorf("wetlab protocol \"%s\" not supported, please choose one of: %s", wetlabProtocol, nanoporeWetlabProtocolOptionsString)
+	}
+	if technology == "Nanopore" && medakaModel == "" {
+		medakaModel = defaultMedakaModel
+	}
+	if clearLabs && technology == "Illumina" {
+		return fmt.Errorf("clearlabs is only supported for Nanore")
```
Comment (tfrcarvalho) [inline cmd/consensusGenome/consensusGenome.go:159]: s/Nanore/Nanopore/

**Review by tfrcarvalho — APPROVED** [review]
Looks good given my limited knowledge of Go.

**Comment by tfrcarvalho** [comment]
What is the plan for codecov constraints? Are you updating tests?

File: cmd/consensusGenome/consensusGenome.go (hunk 1 above)
Comment (morsecodist) [reply cmd/consensusGenome/consensusGenome.go:29]: I had the same thought. It was hard-coded in the web client so I hard-coded it here. On one hand it is a huge pain but on the other these are listed in the help text and I am not sure we should require network connectivity for that. The hardcoding may be the lesser of two evils.

File: cmd/consensusGenome/consensusGenome.go [hunk 3]
```diff
@@ -59,11 +98,36 @@ func loadSharedFlags(c *cobra.Command) {
...
+		strings.Join(util.StringMapKeys(nanoporeWetLabProtocols), "\", \""),
+	)
+
+	c.Flags().StringVarP(&projectName, "project", "p", "", "Project name. Make sure the project is created on the website (required)")
 	c.Flags().StringToStringVarP(&stringMetadata, "metadatum", "m", map[string]string{}, "metadatum name and value for your sample, ex. 'host=Human'")
 	c.Flags().StringVar(&metadataCSVPath, "metadata-csv", "", "Metadata local file path.")
 	c.Flags().StringVar(&technology, "sequencing-platform", "", fmt.Sprintf("Sequencing platform used to sequence the sample, options: %s", technologyOptionsString))
-	c.Flags().StringVar(&wetlabProtocol, "wetlab-protocol", "", fmt.Sprintf("Wetlab protocol followed (only supported/required for illumina), options: %s", wetlabProtocolOptionsString))
+	c.Flags().StringVar(&wetlabProtocol, "wetlab-protocol", "", fmt.Sprintf(
+		"Wetlab protocol followed.\n  Options for Nanopore (optional, default: \"%s\"): %s\n  Options for Illumina (reqired): %s",
```
Comment (j-x-han) [inline cmd/consensusGenome/consensusGenome.go:116]: nit: `required`

**Comment by morsecodist** [comment]
> What is the plan for codecov constraints? Are you updating tests?

Added new tests

**Pull request merged**